### PR TITLE
feat: Use new language-specific settings in service config YAML

### PR DESCRIPTION
**Pull request by jskeet**
This significantly reduces the amount of customization needed in google-cloud-dotnet.

Note: it's expected that this won't build yet, as the options aren't available yet...

### CODE REVIEW

**Comment by jskeet** [comment]
Note to self: the snippet metadata generation for service.shortName still needs fixing up.

**Comment by jskeet** [comment]
This is now ready for review (assuming it goes green), one commit at a time.

**Review by amanda-tarafa — APPROVED** [review]
LGTM, this looks really good!

File: Google.Api.Generator.Tests/ProtoTests/PublishingSettings/CommonResourceFakes.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿// Copyright 2019 Google Inc. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (jskeet) [reply Google.Api.Generator.Tests/ProtoTests/PublishingSettings/CommonResourceFakes.cs:1]: Nope, not deliberate - copy paste error :) Will fix before merging; thanks for spotting.

[the remark this answers is not in the document]
Comment (jskeet) [reply Google.Api.Generator.Tests/ProtoTests/PublishingSettings/CommonResourceFakes.cs]: Now fixed (and using file-scoped namespaces). Will merge on green.

**Comment by jskeet** [comment]
Closing and reopening to hopefully kickstart the build...

**Pull request merged**